reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java

```diff
@@ -106,7 +106,14 @@ private static Observation defaultObservation(
 			MicrometerObservationListenerConfiguration configuration,
 			@Nullable Function<ObservationRegistry, Observation> observationSupplier) {
 		if (observationSupplier != null) {
-			return observationSupplier.apply(configuration.registry);
+			try {
+				final Observation observation = observationSupplier.apply(configuration.registry);
+				if (observation != null) {
+					return observation;
+				}
+			} catch (final Exception e) {
+				LOGGER.error("Error creating observation with given supplier, use default supplier instead", e);
```

Comment:
I think it'd be better to not catch the exception, and let the user notice the error. Clearly user's intention is to provide a customiser, while this could come unnoticed until logs are read. My recommendation - remove this latest change and adapt the tests to validate the exception is thrown during assembly time.

Reply:
Hi @chemicL , ok, I've removed the catch and modified the test to show the exception. I hope it's what you wanted

Reply:
Thank you. I just requested a javadoc update and that will be it from me.